### PR TITLE
Dev tooling

### PR DESCRIPTION
**Pull request by jtraband**
Added browser-sync support, 
Added make-example mixin ( similar to makeTabs mixin)
Updated doc-shredder shredSingleDir to return a promise ( needed for browser-sync)
Updated readme.

### CODE REVIEW

File: public/doc-shredder/regionExtractor.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ module.exports = function regionExtractor() {
       }
     });
 
+    var rx = new RegExp(nullLine + '\n', 'g');
```
Comment (btford) [inline public/doc-shredder/regionExtractor.js]: Hoist this up to the top, and give it a mote descriptive name

**Comment by btford** [comment]
LGTM :+1: Merge away

**Comment by naomiblack** [comment]
CLA is signed by IdeaBlade

**Comment by naomiblack** [comment]
When I run `harp compile` I get a jade error. Looks like a missing file?

```
{
  "source": "Jade",
  "dest": "HTML",
  "filename": "/Users/naomib/github/angular.io/public/doc-shredder/test/test_jade/setupAlt.jade",
  "lineno": 1,
  "name": "Error",
  "message": "ENOENT, no such file or directory '/Users/naomib/github/angular.io/_includes/_util-fns.jade'",
```

**Comment by naomiblack** [comment]
Still seeing this issue. Is it possible there's a file you forgot to add to the commit?

```
$ harp compile
{
  "source": "Jade",
  "dest": "HTML",
  "filename": "/Users/naomib/github/angular.io/public/doc-shredder/test/test_jade/setupAlt.jade",
  "lineno": 1,
  "name": "Error",
  "message": "ENOENT, no such file or directory '/Users/naomib/github/angular.io/_includes/_util-fns.jade'",
```

**Pull request merged**